### PR TITLE
Add cert-expiry to swarm update

### PR DESCRIPTION
**Pull request by tonistiigi**
- Add cert-expiry to swarm update
- Add integration test for parsing swarm update options

@diogomonica @vieux

### CODE REVIEW

**Comment by vieux** [comment]
LGTM ping @dnephin

**Comment by icecrime** [comment]
LGTM

**Comment by diogomonica** [comment]
This LGTM, but @tonistiigi do you mind adding a test that attempts at updating the duration to anything under `30m`? I want to make sure we have a test for that failure case (the remote api should reject it).

**Comment by thaJeztah** [comment]
looks like this needs docs updates as well?

File: api/client/swarm/update.go [hunk 1]
```diff
@@ -38,6 +39,7 @@ func newUpdateCommand(dockerCli *client.DockerCli) *cobra.Command {
 	flags.StringVar(&opts.secret, "secret", "", "Set secret value needed to accept nodes into cluster")
 	flags.Int64Var(&opts.taskHistoryLimit, "task-history-limit", 10, "Task history retention limit")
 	flags.DurationVar(&opts.dispatcherHeartbeat, "dispatcher-heartbeat", time.Duration(5*time.Second), "Dispatcher heartbeat period")
+	flags.DurationVar(&opts.nodeCertExpiry, "cert-expiry", time.Duration(90*24*time.Hour), "Duration node certificates will be valid for")
```
Comment (aaronlehmann) [inline api/client/swarm/update.go]: How about "Validity period for node certificates"?

**Comment by tonistiigi** [comment]
@diogomonica @thaJeztah updated

**Comment by diogomonica** [comment]
Looks great. Thanks for picking this up so fast. <3

**Comment by thaJeztah** [comment]
docs LGTM

**Comment by thaJeztah** [comment]
docs CI failure can be ignored (known issue with those links)

**Comment by aaronlehmann** [comment]
Shouldn't this flag be supported in `swarm init` as well?

**Comment by thaJeztah** [comment]
ping @albers / @mgoelzer for bash completion

**Comment by tonistiigi** [comment]
> Shouldn't this flag be supported in swarm init as well?

I agree but all non-acceptance related fields are missing from there(not even supported by API). So this should be another change.

**Comment by icecrime** [comment]
@tonistiigi Failure on win2lin:


```
19:21:42 FAIL: c:/Jenkins/gopath/src/github.com/docker/docker/integration-cli/docker_cli_swarm_test.go:12: DockerSwarmSuite.TestSwarmUpdate
19:21:42 
19:21:42 c:/Jenkins/gopath/src/github.com/docker/docker/integration-cli/docker_cli_swarm_test.go:13:
19:21:42     d := s.AddDaemon(c, true, true)
19:21:42 github.com/docker/docker/integration-cli/_test/_obj_test/daemon.go:154:
19:21:42     ...open github.com/docker/docker/integration-cli/_test/_obj_test/daemon.go: The system cannot find the path specified.
19:21:42 ... value *exec.Error = &exec.Error{Name:"dockerd", Err:(*errors.errorString)(0xc082022f40)} ("exec: \"dockerd\": executable file not found in %PATH%")
19:21:42 ... [d735700] could not find docker binary in $PATH
```

**Pull request merged**